docs/DevGuide/Amr.md

```diff
@@ -0,0 +1,367 @@
+the derived classes).  In particular, each criterion should compute an
+array of amr::Flag%s that represent the recommended refinemenent
+choice in each logical dimension.  These choices can be computed from
+any items in the db::DataBox of the element or additional compute
+items specified by the criterion.  It is expected that a criterion
+will compute something and then either choose to recommend refinement,
+no change, or coarsening the element either in size (h-refinement) or
+polynomial order (p-refinement).  A criterion does not need to handle
+anything that is enforced by the amr::Policies such as worry about
+bounds on the refinement level or number of grid points.
+
+### AMR Policies {#dev_guide_amr_policies}
+
+The current AMR algorithm has a set of rules that are enforced after
+the overall refinement choice is determined from the AMR criteria.
+
+The following rules simplify the code:
+
+- An Element cannot join if it is splitting in any dimension.  Instead
+  any amr::Flag::Join is changed to amr::Flag::DoNothing. Relaxing
+  this restriction would be a non-trivial change that would require
+  more complicated logic in determining if neighboring elements can
+  join, as well as an additional code path to handle the creation and
+  initialization of the new elements. This should not be a significant
+  restriction as in most cases when this would occur it is unlikely
+  that the siblings of the Element would agree to a valid join.
+
+- In two or three spatial dimensions, there must be a 2:1 balance
+  (i.e. within one refinement level) between neighbors in the
+  dimension(s) parallel to the face of the element.  Relaxing this
+  restriction would be a significant change to the code as it as the
+  code for communicating boundary corrections assumes 2:1 balance.  In
+  order to maintain 2:1 balance with neighboring elements, a choice of
+  amr::Flag::Join may be changed to amr::Flag::DoNothing, and
+  amr::Flag::DoNothing may be changed to amr::Flag::Split.
+
+The following policies are among those controlled by input file
+options (see \ref dev_guide_amr_input_options "Controlling AMR" and
+the documentation for amr::Policies):
+
+- Whether to do isotropic or anisotropic refinement.  If isotropic
+  refinement is done, the AMR decision in each dimension are changed
+  to the decision of the dimension with the highest priority amr::Flag
+  (i.e. amr::Flag::Split has the highest priority).
+
+- Whether or not to enforce 2:1 balance for the refinement level in
+  the direction perpendicular to the face.
+
+- The minimum and maximum of the refinement level and the number of
+  grid points.  The actual bounds on these will be the stricter of
+  those specified by the input file and those set by the code.  For
+  the number of grid points, the code bounds are set by the
+  Spectral::Basis of the Mesh.  For the refinement level, the minimum
+  is zero, and the maximum is set by
+  `ElementId::max_refinement_level`.
```

Comment:
This link didn't show up. Maybe because `max_refinement_level` isn't documented? Could just link `ElementId` instead.